### PR TITLE
[KYUUBI #5877][FOLLOWUP] Add spark output mode to better support PySpark notebook

### PR DESCRIPTION
**Pull request by turboFei**
# :mag: Description
## Issue References 🔗

This pull request fixes #

## Describe Your Solution 🔧

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Types of changes :bookmark:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Related Unit Tests

---

# Checklists
## 📝 Author Self Checklist

- [ ] My code follows the [style guidelines](https://kyuubi.readthedocs.io/en/master/contributing/code/style.html) of this project
- [ ] I have performed a self-review
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] This patch was not authored or co-authored using [Generative Tooling](https://www.apache.org/legal/generative-tooling.html)

## 📝 Committer Pre-Merge Checklist

- [ ] Pull request title is okay.
- [ ] No license issues.
- [ ] Milestone correctly set?
- [ ] Test coverage is ok
- [ ] Assignees are selected.
- [ ] Minimum number of approvals
- [ ] No changes are requested

**Be nice. Be informative.**

### CODE REVIEW

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala [hunk 1]
```diff
@@ -3258,6 +3258,14 @@ object KyuubiConf {
       .booleanConf
       .createWithDefault(true)
 
+  val ENGINE_SPARK_NOTEBOOK_MODE: ConfigEntry[Boolean] =
+    buildConf("kyuubi.engine.spark.notebook.mode")
+      .doc("Enable this to support notebook well." +
+        " For python notebook, the `text/plain` wrapped output will be returned.")
```
Comment (pan3793) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:3264]: I'm not quite familiar with this area, could you please elaborate more about the difference between this configuration true/false?

Comment (pan3793) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:3264]: from the logic of the code, maybe we can name it as `kyuubi.engine.spark.output.mode` with options "auto" and "notebook".

cc @cfmcgrady do you have suggestions?

Comment (turboFei) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:3264]: for notebook mode, for the output:
```
{ "text/plain": <content>}
```

for non-notebook mode, the output:
```
<content>
```

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala [hunk 2]
```diff
@@ -3258,13 +3258,19 @@ object KyuubiConf {
...
-        " For python notebook, the `text/plain` wrapped output will be returned.")
+  object EngineSparkOutputMode extends Enumeration {
+    type EngineSparkOutputMode = Value
+    val AUTO, NOTEBOOK = Value
+  }
+
+  val ENGINE_SPARK_OUTPUT_MODE: ConfigEntry[String] =
+    buildConf("kyuubi.engine.spark.output.mode")
+      .doc("The output mode of Spark engine: <ul>" +
+        " <li>AUTO: For pySpark, the extracted `text/plain` from python response as output.</li>" +
```
Comment (pan3793) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:3269]: **PySpark**: https://spark.apache.org/docs/latest/api/python/index.html

**Review by pan3793 — APPROVED** [review]

**Comment by turboFei** [comment]
thanks, merged to 1.9

**Pull request closed without merging**